common/pipelinegen/config_builder.go

```diff
@@ -257,6 +257,14 @@ func insertServiceGraphPipeline(currentConfig *config.Config) {
 	}
 	pipeline.Exporters = append(pipeline.Exporters, consts.ServiceGraphConnectorName)
 	currentConfig.Service.Pipelines[rootPipelineName] = pipeline
+
+	// Add ServiceGraphConnectorName receiver to every prometheus pipeline starts with metrics/prometheus-
+	for pipelineName, pipeline := range currentConfig.Service.Pipelines {
+		if strings.HasPrefix(pipelineName, "metrics/prometheus-") {
+			pipeline.Receivers = append(pipeline.Receivers, consts.ServiceGraphConnectorName)
```

Comment:
This assumes the connector is in the config, because metrics always enable traces, and traces enables service graph, right?
I think if one will disable the service graph in config, this part will make collector fail to load (probably not likely but supported).
wondering if we can assert that connector exist for safety for whatever scenario we might hit.